src/main/java/org/gridsuite/directory/server/DirectoryService.java

```diff
@@ -228,6 +230,69 @@ private void emitDirectoryChanged(UUID directoryUuid, String elementName, String
             .then();
     }
 
+    public void updateElementDirectory(UUID elementUuid, UUID newDirectoryUuid, String userId) {
+        Optional<DirectoryElementEntity> optElement = getElementEntity(elementUuid);
+        Optional<DirectoryElementEntity> optNewDirectory = getElementEntity(newDirectoryUuid);
+        DirectoryElementEntity oldDirectory;
+        DirectoryElementEntity element;
+        DirectoryElementEntity newDirectory;
+        if (optElement.isEmpty()) {
+            throw DirectoryException.createElementNotFound(ELEMENT, elementUuid);
+        }
+        if (optNewDirectory.isEmpty()) {
+            throw DirectoryException.createElementNotFound(DIRECTORY, newDirectoryUuid);
+        }
+        element = optElement.get();
+        newDirectory = optNewDirectory.get();
+
+        if (element.getType().equals(DIRECTORY)) {
+            throw new DirectoryException(IS_DIRECTORY);
+        }
+        if (!newDirectory.getType().equals(DIRECTORY)) {
+            throw new DirectoryException(NOT_DIRECTORY);
+        }
+        if (!isElementUpdatable(toElementAttributes(element), userId, false)) {
+            throw new DirectoryException(NOT_ALLOWED);
+        }
+        if (directoryHasElementOfNameAndType(newDirectoryUuid, userId, element.getName(), element.getType())) {
+            throw new DirectoryException(NOT_ALLOWED);
+        }
+
+        oldDirectory = getElementEntity(element.getParentId()).get();
+        if (!newDirectory.getIsPrivate().equals(oldDirectory.getIsPrivate())) {
+            throw DirectoryException.createDirectoryWithDifferentAccessRights(elementUuid, newDirectoryUuid);
+        }
+
+        element.setParentId(newDirectoryUuid);
+        directoryElementRepository.save(element);
+
+        emitDirectoryChanged(
+            element.getParentId(),
+            element.getName(),
+            userId,
+            isPrivateForNotification(element.getParentId(), false),
+            isRootDirectory(element.getId()),
+            NotificationType.UPDATE_DIRECTORY
+        );
+
+        emitDirectoryChanged(
+            oldDirectory.getId(),
+            element.getName(),
+            userId,
+            isPrivateForNotification(element.getParentId(), false),
+            isRootDirectory(element.getId()),
```

Comment:
We call this isRootDirectory here with an element which is not of DIRECTORY type... well later it will work for any type ok. Maybe we should check the type in `isRootDirectory` then ?